### PR TITLE
[library] Move library to `vernac`

### PR DESCRIPTION
**Pull request by ejgallego**
As suggested by Maxime, we move library to vernac.

This PR is just a proof-of-concept and more work would be needed to be mergeable.

cc: @maximedenes

### CODE REVIEW

**Comment by ejgallego** [comment]
It seems that moving `Declaremods` and even `Nametab` to `interp` would be feasible; but there is at least a very big hurdle which is the use of `Namegen`.

File: vernac/library.mllib [hunk 1]
```diff
@@ -7,7 +7,6 @@ Global
 Decl_kinds
```
Comment (SkySkimmer) [inline vernac/library.mllib]: Is moving this file intended?

**Comment by SkySkimmer** [comment]
I'm a bit confused, you're only moving the `library.ml` not all of `library/` right?

File: vernac/library.mllib (hunk 1 above)
Comment (ejgallego) [reply vernac/library.mllib]: Yes, this is already done in one of the proof PRs.

Comment (ejgallego) [reply vernac/library.mllib]: Required quite a bit of untangling.

Comment (SkySkimmer) [reply vernac/library.mllib]: I mean moving the mllib in this PR, not decl_kinds in the future.

Comment (ejgallego) [reply vernac/library.mllib]: Ah no, that was mistake sorry.

**Comment by ejgallego** [comment]
> I'm a bit confused, you're only moving the `library.ml` not all of `library/` right?

Yes sorry for the mllib problem; indeed I am only moving `Library`, let me amend.

**Comment by ejgallego** [comment]
This PR was just a draft to illustrate how far we are; IMHO no point on keeping it open, but can still be useful for those curious.

**Pull request closed without merging**